src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/jpa/repository/UserRepository.kt

```diff
@@ -0,0 +1,16 @@
+package uk.gov.justice.digital.hmpps.externalusersapi.jpa.repository
+
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.CrudRepository
+import uk.gov.justice.digital.hmpps.externalusersapi.model.User
+import uk.gov.justice.digital.hmpps.externalusersapi.security.AuthSource
+import java.util.Optional
+import java.util.UUID
+
+interface UserRepository : CrudRepository<User, UUID>, JpaSpecificationExecutor<User> {
+  @Query("select distinct u from User u left join fetch u.authorities where u.username = :username and u.source = :source")
```

Comment:
In our context AuthSource will always be external. We should be able to remove the AuthSource enum and just hard code the filter in the query for now?

Reply:
The actual method invoking has Authsource hardcoded as 
'findByUsernameAndSource(username, AuthSource.auth)'